keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -28,33 +30,76 @@ status: implementable
+* [Risks and Mitigations](#risks-and-mitigations)
 * [Graduation Criteria](#graduation-criteria)
 * [Implementation History](#implementation-history)
 
 ## Summary
 
-Kustomize wants to generate Kubernetes Secret objects from general key-value pairs where the value is supposed to be a secret. Currently Kustomize only supports reading secret values from local files which raises security concerns about file lifetime and access. Reading secret values from the execution of arbitrary "commands" in a kustomization.yaml file introduces concerns in a world where kustomization files can be used directly from the internet when a user runs `kustomize build`.
-
-This proposal describes obtaining secret values safely using golang [plugins](https://golang.org/pkg/plugin/).
+Kustomize users want to generate Kubernetes Secret
+objects from general key:value (KV) pairs where the
+value is supposed to be a secret.  Users want to
+generate these pairs through integration with secret
+management tools (e.g. see comments on
+[692][execRemoval]).
+
+Currently Kustomize only supports reading secret values
+from local files which raises security concerns about
+file lifetime and access. Reading secret values from
+the execution of arbitrary "commands" in a
+kustomization.yaml file introduces concerns in a world
+where kustomization files can be used directly from the
+internet when a user runs `kustomize build`.  This
+proposal describes the syntax for a new key:value
+generator plugin framework supporting an arbitrary
+number of plugin types to generate key:value pairs.
 
 ## Motivation
 
-Not having a way to integrate `SecretGenerator` with secret management tools requires hacky workarounds.
+Not having a way to integrate `SecretGenerator` with
+secret management tools requires hacky, insecure
+workarounds.
 
 ### Goals
 
-- Give users a way to intergrate `SecretGenerator` with secret management tools in a safe way.
+- In the `GeneratorArgs` section of a kustomization
+  file, a user may specify a plugin type, and a
+  specific instance of that type, for generating
+  key:value pairs.
+
+- The specification will allow for any number of plugin
+  types, and any number of instances of those types.
+  
+- The first type supported will be
+  [goplugins](https://golang.org/pkg/plugin), to enable
+  kustomize source code contributors to add custom KV
+  generators without the need to maintain a kustomize
+  source code fork.
+  
+  Kustomize maintainers expect developers who use a
+  goplugin to understand that a kustomize binary and
+  any goplugins expected to work with it must be
+  compiled on the same machine with the same compiler
+  against the same set of (transitive) libraries.
+  _It's the developers responsibility to bundle the
+  binary and shared libraries together in an immutable
+  container for actual use._
+
+- Other kinds of plugins, e.g. an _execute this binary_
+  plugin, should be subsequently easy to add, and could
+  be appropriate for end user use.
```

Comment:
Just to be safe, add a line about "being careful to avoid pitfalls of exec removal" or "following similar constraints to go plugins" or something.

Reply:
done - reiterated that they'd need a KEP